frontend/src/v5/store/drawings/drawings.sagas.ts

```diff
@@ -59,7 +58,46 @@ export function* fetchDrawingStats({ teamspace, projectId, drawingId }: FetchDra
 	}
 }
 
+export function* fetchCategories({ teamspace, projectId }: FetchCategoriesAction) {
+	try {
+		const categories = yield API.Drawings.fetchCategories(teamspace, projectId);
+		yield put(DrawingsActions.fetchCategoriesSuccess(projectId, categories));
+
+	} catch (error) {
+		yield put(DialogsActions.open('alert', {
+			currentActions: formatMessage({ id: 'drawings.fetchCategories.error', defaultMessage: 'trying to fetch categories' }),
+			error,
+		}));
+	}
+}
+
+export function* createDrawing({ teamspace, projectId, drawing, onSuccess, onError }: CreateDrawingAction) {
+	try {
+		const id = yield API.Drawings.createDrawing(teamspace, projectId, drawing);
+		const newDrawing = { _id: id, ...drawing };
+		yield put(DrawingsActions.createDrawingSuccess(projectId, newDrawing));
+
+		onSuccess();
+	} catch (error) {
+		onError(error);
+	}
+}
+
+export function* updateDrawing({ teamspace, projectId, drawingId, drawing, onSuccess, onError }: UpdateDrawingAction) {
+	try {
+		yield API.Drawings.updateDrawing(teamspace, projectId, drawingId, drawing);
+		const updatedDrawing = { _id: drawingId, ...drawing };
```

Comment:
If you're updating a drawing this means that the drawing object should already contain an _id, right? Can't you just use `drawing` here and remove `drawingId` from this saga's props using `drawing._id` instead. Same for the API function

Reply:
I changed this to match container's standard  which get passed a Partial<container> and the contianerid